### PR TITLE
*BSD support

### PR DESCRIPTION
**Pull request by dasehak**

### CODE REVIEW

File: src/Common/PlatformBSD.inl [hunk 1]
```diff
@@ -14,16 +14,18 @@
 #include <sched.h>
 
 #include <algorithm> // for min max
+#include <stddef.h>
 
 #include <string>
-#include <pthread_np.h>
 #include <pthread.h>
-#define pthread_setname_np pthread_set_name_np
```
Comment (Xottab-DUTY) [inline src/Common/PlatformBSD.inl:21]: https://discord.com/channels/410170555619082240/489453264157409281/1051924182495154287
image

File: src/Layers/xrRender/ParticleEffect.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #endif
 #endif
 
-#if defined(XR_PLATFORM_LINUX) || defined(XR_PLATFORM_APPLE) // XXX: remove
+#if defined(XR_PLATFORM_LINUX) || defined(XR_PLATFORM_APPLE) || defined(XR_PLATFORM_BSD) // XXX: remove
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/ParticleEffect.cpp]: @dasehak, можешь везде по файлам проставить, чтобы порядок был такой: `Linux || BSD || Apple`?

Comment (dasehak) [reply src/Layers/xrRender/ParticleEffect.cpp]: Без проблем.

File: src/xrCore/xrDebug.cpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ static BOOL bException = FALSE;
 #   if __has_include(<execinfo.h>)
 #       include <execinfo.h>
 #   endif
-#elif defined(XR_PLATFORM_APPLE)
+#elif defined(XR_PLATFORM_APPLE) || defined(XR_PLATFORM_BSD)
```
Comment (Xottab-DUTY) [inline src/xrCore/xrDebug.cpp]:
```suggestion
#elif defined(XR_PLATFORM_BSD) || defined(XR_PLATFORM_APPLE)
```

File: src/Layers/xrRenderPC_GL/gl_rendertarget_build_textures.cpp [hunk 4]
```diff
@@ -46,6 +46,10 @@ void CRenderTarget::build_textures()
     }*/
     // Build material(s)
     {
+        #ifdef XR_PLATFORM_BSD
+        fedisableexcept(FE_UNDERFLOW | FE_INEXACT); // I really want to see a better solution
+        #endif
+
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderPC_GL/gl_rendertarget_build_textures.cpp]:
```suggestion
```

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
I removed a bit of changes from this pull request so that we can merge what is ready to merge.
Some changes were pushed directly to dev with of refactoring of the surrounding code. (6440756842c228027d46cc63a70cfb0e53b141c8 and c85e14b8e4a9c4083d7117f9dcc422e3e9800fb1)

Just in case, sorry for the inconvenience :D

**Comment by Xottab-DUTY** [comment]
And, of course, thank you very much!

**Pull request merged**